### PR TITLE
Add min exponent width option in double-to-string shortest conversion

### PR DESCRIPTION
**Pull request by pengzhenyi2015**
Prefix 0 is added when call golang fmt.Sprint() to convert double to shortest string
For example: 0.00001 is convert to 1e-05, but not 1e-5

In some situations, I want to convert double to shortest string in "golang format", so a "min exponent width option" will be useful

### CODE REVIEW

**Comment by pengzhenyi2015** [comment]
@googlebot I signed it!)

File: double-conversion/double-to-string.h [hunk 1]
```diff
@@ -111,7 +111,8 @@ class DoubleToStringConverter {
                           int decimal_in_shortest_low,
                           int decimal_in_shortest_high,
                           int max_leading_padding_zeroes_in_precision_mode,
-                          int max_trailing_padding_zeroes_in_precision_mode)
+                          int max_trailing_padding_zeroes_in_precision_mode,
+                          int min_exponent_width = 0)
```
Comment (floitsch) [inline double-conversion/double-to-string.h]: Also add a line (or two) to the comment of the constructor, explaining what the parameter does.
Shouldn't need a lot of explanation (quite self-explanatory), but mention that the exponent will never have more than 5 digits.
Then also add a comment to the constant (`5`) in double-to-string.cc and state that changing this constant also requires updating the comment in double-to-string.h

File: test/cctest/test-conversions.cc [hunk 2]
```diff
@@ -71,6 +71,36 @@ TEST(DoubleToShortest) {
...
+  CHECK_EQ("111111", builder.Finalize());
+
+  builder.Reset();
+  CHECK(dcExpWidth.ToShortest(10000000000.0, &builder));
+  CHECK_EQ("1e+10", builder.Finalize());
+
+  builder.Reset();
+  CHECK(dcExpWidth.ToShortest(1000000000.0, &builder));
+  CHECK_EQ("1e+09", builder.Finalize());
+
```
Comment (floitsch) [inline test/cctest/test-conversions.cc]: Add a few more tests with corner-cases:
- min_exponent_width set to 0. (which is the default, but still)
- min_exponent_width set to 100.

**Comment by floitsch** [comment]
Thanks.

File: test/cctest/test-conversions.cc (hunk 2 above)
Comment (pengzhenyi2015) [reply test/cctest/test-conversions.cc]: The corner-cases are added

File: double-conversion/double-to-string.h (hunk 1 above)
Comment (pengzhenyi2015) [reply double-conversion/double-to-string.h]: Thanks for your advice, I added some comments to introduce this option

**Comment by pengzhenyi2015** [comment]
@floitsch Thanks for your advice

File: double-conversion/double-to-string.h [hunk 3]
```diff
@@ -104,14 +104,18 @@ class DoubleToStringConverter {
   //   ToPrecision(230.0, 2) -> "230"
   //   ToPrecision(230.0, 2) -> "230."  with EMIT_TRAILING_DECIMAL_POINT.
   //   ToPrecision(230.0, 2) -> "2.3e2" with EMIT_TRAILING_ZERO_AFTER_POINT.
+  //
+  // The exponent width may add up to min_exponent_width by adding prefix '0',
+  // but the exponent will never have more than 5 digits as kMaxExponentLength is 5
```
Comment (floitsch) [inline double-conversion/double-to-string.h]: No need to mention `kMaxExponentLength`. It's an internal detail.
Simply:
"When min_exponent_width is used for exponential representations. The converter adds leading '0's, to the exponent until the exponent is at least min_exponent_width digits long.
The min_exponent_width is clamped to 5. As such, the exponent may never have more than 5 digits in total."

Comment (floitsch) [reply double-conversion/double-to-string.h]: or something similar.

**Comment by pengzhenyi2015** [comment]
The comment is updated

File: double-conversion/double-to-string.h [hunk 4]
```diff
@@ -104,14 +104,21 @@ class DoubleToStringConverter {
   //   ToPrecision(230.0, 2) -> "230"
   //   ToPrecision(230.0, 2) -> "230."  with EMIT_TRAILING_DECIMAL_POINT.
   //   ToPrecision(230.0, 2) -> "2.3e2" with EMIT_TRAILING_ZERO_AFTER_POINT.
+  //
+  // When min_exponent_width is used for exponential representations.
```
Comment (floitsch) [inline double-conversion/double-to-string.h]:
```suggestion
  // The min_exponent_width is used for exponential representations.
```

**Comment by floitsch** [comment]
One last tiny nit left.

**Comment by floitsch** [comment]
Thanks.

**Review by Beeloo0011 — COMMENTED** [review]
- [remove]
